hyperspy/_components/expression.py

```diff
@@ -215,3 +218,14 @@ def compile_function(self, module="numpy", position=False):
                         self,
                         Expression)
                     )
+
+    def is_linear(self, expr, vars):
```

Comment:
I would move this function out of the class.

Also I don't think it needs to be as general as it is in the stackexchange example:
```python
def is_linear(expression, name):
    try:
        return sympy.Eq(sympy.diff(expression, name, 2), 0)
    except TypeError:
        return False